jerry-core/api/jerry.c

```diff
@@ -4602,6 +4602,39 @@ jerry_set_realm (jerry_value_t realm_value) /**< jerry api value */
 #endif /* ENABLED (JERRY_BUILTIN_REALMS) */
 } /* jerry_set_realm */
 
+/**
+ * Gets the 'this' binding of a realm
+ *
+ * @return thrown error - if any error is happened
+ *         this value - otherwise
+ */
+jerry_value_t
+jerry_realm_get_this (jerry_value_t realm_value) /**< realm value */
+{
+  jerry_assert_api_available ();
+
+#if ENABLED (JERRY_BUILTIN_REALMS)
+  if (ecma_is_value_object (realm_value))
+  {
+    ecma_object_t *object_p = ecma_get_object_from_value (realm_value);
+
+    if (ecma_get_object_is_builtin (object_p)
+        && ecma_builtin_is_global (object_p))
+    {
+      ecma_global_object_t *global_object_p = (ecma_global_object_t *) object_p;
+
+      ecma_ref_object (ecma_get_object_from_value (global_object_p->this_binding));
+      return global_object_p->this_binding;
+    }
+  }
+
+  return jerry_throw (ecma_raise_type_error (ECMA_ERR_MSG ("Passed argument is not a realm")));
+#else /* !ENABLED (JERRY_BUILTIN_REALMS) */
+  JERRY_UNUSED (realm_value);
+  return jerry_get_global_object ();
```

Comment:
Afaik, this behaviour should be documented. That is if the realms are disabled the global object is returned.